app/models/enterprise.rb

```diff
@@ -118,7 +118,7 @@ class Enterprise < ActiveRecord::Base
       except(:select).
       select('DISTINCT enterprises.id')
 
-    if ready_enterprises.present?
+    if ready_enterprises.any?
```

Comment:
`any?` will perform a COUNT thus, the `else` branch at line 124 seems pointless to me.